lib/internal/fs/streams.js

```diff
@@ -243,6 +269,34 @@ function WriteStream(path, options) {
   // For backwards compat do not emit close on destroy.
   options.emitClose = false;
 
+  if (options.open && typeof options.open !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.open', 'function', options.open);
+  }
+
+  if ((options.writev || options.write) &&
+    typeof options.write !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.write', 'function', options.write);
+  }
+
+  if (options.writev && typeof options.writev !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.writev', 'function',
+                                   options.writev);
+  }
+
+  if (options.close && typeof options.close !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.close', 'function', options.close);
+  }
+
+  this[kOpen] = options.open;
+  this[kWrite] = options.write;
+  this[kWritev] = options.writev;
+  this[kClose] = options.close;
```

Comment:
I would prefer these to be _always assigned_, even when using the 'real'  fs.

Reply:
The problem is that (e.g. test) will monkey patch `fs` **after** the stream has been created.